src/ert/_c_wrappers/enkf/subst_config.py

```diff
@@ -1,114 +1,84 @@
-        self, config_content=None, config_dict=None, num_cpu: Optional[int] = None
+        self,
+        defines: Dict[str, str],
+        data_kw: Dict[str, str],
+        runpath_file_name: str,
+        num_cpu: int,
     ):
-        if not (config_content is not None) ^ (config_dict is not None):
-            raise ValueError(
-                "SubstConfig must be instansiated with exactly one"
-                " of config_content or config_dict"
-            )
+        subst_list = SubstitutionList()
 
-        if config_dict is not None:
-            subst_list = SubstitutionList()
-
-            # DIRECTORY #
-            config_directory = config_dict.get(ConfigKeys.CONFIG_DIRECTORY)
-            if isinstance(config_directory, str):
-                subst_list.addItem(
-                    "<CWD>",
-                    config_directory,
-                    "The current working directory we are running from"
-                    " - the location of the config file.",
-                )
-                subst_list.addItem(
-                    "<CONFIG_PATH>",
-                    config_directory,
-                    "The current working directory we are running from"
-                    " - the location of the config file.",
-                )
-            else:
-                raise ValueError(f"{ConfigKeys.CONFIG_DIRECTORY} must be configured")
-
-            # FILE #
-            filename = config_dict.get(ConfigKeys.CONFIG_FILE_KEY)
-            if isinstance(filename, str):
-                subst_list.addItem("<CONFIG_FILE>", filename)
-                subst_list.addItem("<CONFIG_FILE_BASE>", os.path.splitext(filename)[0])
-
-            # CONSTANTS #
-            constants = config_dict.get(ConfigKeys.DEFINE_KEY)
-            if isinstance(constants, dict):
-                for key in constants:
-                    subst_list.addItem(key, constants[key])
-
-            # DATA_KW
-            data_kw = config_dict.get(ConfigKeys.DATA_KW_KEY)
-            if isinstance(data_kw, dict):
-                for key, value in data_kw.items():
-                    subst_list.addItem(key, value)
-
-            # RUNPATH_FILE #
-            runpath_file_name = config_dict.get(
-                ConfigKeys.RUNPATH_FILE, ConfigKeys.RUNPATH_LIST_FILE
-            )
-            runpath_file_path = os.path.normpath(
-                os.path.join(config_directory, runpath_file_name)
-            )
-            subst_list.addItem(
-                "<RUNPATH_FILE>",
-                runpath_file_path,
-                "The name of a file with a list of run directories.",
-            )
-            if num_cpu is not None:
-                subst_list.addItem(
-                    "<NUM_CPU>",
-                    str(num_cpu),
-                    "The number of CPU used for one forward model.",
-                )
+        today_date_string = date.today().isoformat()
+        subst_list.addItem("<DATE>", today_date_string, "The current date.")
```

Comment:
Just a side note the documentation string in this case `The current date.` seems to serve no apparent purpose, the C code keeps track of it for no reason, and it is never printed to file or used in any print to console. So I would just replace it with a comment when the code migrates fully to python

Reply:
ah, excellent - more code that can be removed ^^

> replace it with a comment

how do you mean, replace with a comment? based on your findings, I would just altogether scrap the docstring, I think, but that's not what you had in mind, is it?

Reply:
live in a meeting discussion shows that indeed this docstring stuff can probably be deleted. thanks again for pointing it out!